### PR TITLE
exclude test build files from packages

### PR DESCRIPTION
**Pull request by peter-hamilton**
## Description

Tests aren't being type checked because they are excluded `tsconfig.json` files. If this exclusion is removed type-checking is applied, but the resultant test build files need to be excluded from the package. 

This PR
- adds the `!build/ts/**/*.test.*` file exclusion the to all `package.json` files. `build/*` include pattern was changed to `build` in order for the exclusion pattern to work properly.
- removes test exclusion from `tsconfig.json` in cases where there aren't type errors. Most packages have type errors that will be addressed in the future. To check just remove the test patterns from the exclude section of `tsconfig.json`.
- **Also**, renamed `test/` directories to `tests/` across all packages. This improved consistency removes the need to match multiple patterns for tests.
  - Changes beyond the rename are in [this commit](https://github.com/Shopify/quilt/pull/2005/commits/b99480e3fe8f9c37c182275e89a19f29c778ce38).

To test what files are included in a package, use `npx npm-packlist` from the package directory.

## Type of change

- [X] `in all` Patch: Bug (non-breaking change which fixes an issue)

## Checklist

- [X] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/address-mocks/package.json [hunk 1]
```diff
@@ -31,8 +31,9 @@
     "@shopify/jest-dom-mocks": "^3.0.5"
   },
   "files": [
-    "build/*",
+    "build/",
     "!build/*.tsbuildinfo",
+    "!build/ts/**/*.test.*",
     "index.js",
```
[the remark this answers is not in the document]
Comment (dahukish) [reply packages/address-mocks/package.json:37]: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#files

Unless we are being explicit for clarity.

File: packages/graphql-tool-utilities/tsconfig.json [hunk 2]
```diff
@@ -5,10 +5,5 @@
     "rootDir": "src",
     "baseUrl": "src"
   },
-  "include": [
-    "../../config/typescript/*.ts",
-    "./src/**/*.ts",
-    "./src/**/*.tsx"
-  ],
-  "exclude": ["**/*.test.ts", "**/*.test.tsx"]
+  "include": ["../../config/typescript/*.ts", "./src/**/*.ts", "./src/**/*.tsx"]
```
Comment (dahukish) [inline packages/graphql-tool-utilities/tsconfig.json:8]: could these go in the base config if most packages follow the same pattern?

**Review by dahukish — APPROVED** [review]
LGTM

File: packages/address-mocks/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (peter-hamilton) [reply packages/address-mocks/package.json:37]: That's true. Though I'd lean towards keeping it explicit.

File: .eslintignore [hunk 3]
```diff
@@ -6,9 +6,9 @@ packages/*/build
...
 **/*.esnext
 **/*.mjs
-**/test/fixtures/**/default-types/*.ts
+**/tests/fixtures/**/default-types/*.ts
 
 # Don't lint fixture graphql files as they include schemas
-packages/graphql-config-utilities/src/tests/fixtures/**/*.graphql
-packages/graphql-typescript-definitions/test/fixtures/**/*.graphql
-packages/graphql-validate-fixtures/test/fixtures/**/*.graphql
+packages/graphql-config-utilities/src/testss/fixtures/**/*.graphql
```
Comment (BPScott) [inline .eslintignore:12]: Double s here might be the cause of the linting failures

Comment (peter-hamilton) [reply .eslintignore:12]: Thank you. This was driving me nuts.

**Comment by peter-hamilton** [comment]
> `!build/ts/**/*.test.*` is good for excluding test files. I wonder if we should exclude the whole `test` / `tests` folders, as sometimes those folders contain utility files that are only used within tests: `ls -d packages/*/src/**/test{,s}/** | grep -v '\.test\.'` give you a list of them. Do you think doing `"!build/ts/**/tests/"` instead would be worth it?
> 
> Annoyingly it seems half of our tests live in folders called `test` and half live folders called `tests` (`ls -d packages/*/src/**/tests | wc -l` vs `ls -d packages/*/src/**/test | wc -l`). the Shopify convention tends to be naming tests folders `tests` (3999 usages of tests in web, vs 6 for test). That's pretty annoying as ignoring both folder names in package.json feels rubbish. Renaming all instances for `test` folders to `tests` is also gonna be noisy but I'm a sucker for conventions. Reckon it might be worth it?

`test` folders have been renamed to `tests`.

**Review by BPScott — COMMENTED** [review]
Looking pretty great!

I think I spotted two little things to tweak inline. fixup those two and I think we're good to go!

I see there's till many mentions of `**/test/**` in tsconfig files but I guess those are the problematic packages and those entire exclude declarations shall be removed in follow up PRs

File: packages/storybook-a11y-test/tsconfig.json [hunk 4]
```diff
@@ -4,6 +4,5 @@
     "outDir": "build/ts",
     "rootDir": "src",
     "baseUrl": "src"
-  },
-  "include": ["./src/**/*.ts"]
```
[the remark this answers is not in the document]
Comment (peter-hamilton) [reply packages/storybook-a11y-test/tsconfig.json]: Reverted.

**Review by BPScott — APPROVED** [review]

**Pull request merged**